common/thorhelper/thorcommon.hpp

```diff
@@ -281,15 +281,29 @@ class ActivityTimer
         else
             startCycles = 0;
     }
-
-    ~ActivityTimer()
+    inline void flushTimes(cycle_t nowCycles)
+    {
+        accumulator.endCycles = nowCycles;
+        accumulator.totalCycles += (nowCycles - startCycles);
+    }
+    inline void flushTimesIntermittantly()
     {
         if (enabled)
         {
             cycle_t nowCycles = get_cycles_now();
-            accumulator.endCycles = nowCycles;
-            cycle_t elapsedCycles = nowCycles - startCycles;
-            accumulator.totalCycles += elapsedCycles;
+            if (nowCycles - (cycle_t) startCycles > defaultFlushTimerInterval)
```

Comment:
startCycles is unsigned. cycle_t types is signed and in most places we are using cycle_t.  We can make nowCycles unsigned but it'd be better to make startCycles cycle_t type. Shall I change startCycles to cycle_t type so that it is consistant and casting is not necessary?

Reply:
ok, we didn't cast before, but I thought startCycles was a cycle_t, I see it is not now in this context, but is in other classes in this header (e.g. the startTimer member of ActivityTimeAccumulator).
I would make startCycles a cycle_t for consistency.